src/renderer.ts

```diff
@@ -137,19 +137,20 @@ export class Renderer {
     context.stroke();
   }
 
-  constructor(elementId: string | HTMLElement, backend: number) {
-    this.elementId = elementId;
-    if (this.elementId === undefined) {
+  constructor(elementId: string | HTMLCanvasElement, backend: number) {
+    if (!elementId) {
       throw new RuntimeError('BadArgument', 'Invalid id for renderer.');
     }
 
-    this.element = document.getElementById(elementId as string) as HTMLCanvasElement;
-    if (!this.element) this.element = elementId as HTMLCanvasElement;
+    this.element =
+      typeof elementId === 'string' && typeof document !== 'undefined'
+        ? (document.getElementById(elementId as string) as HTMLCanvasElement)
+        : (elementId as HTMLCanvasElement);
 
     // Verify backend and create context
     this.backend = backend;
     if (this.backend === Renderer.Backends.CANVAS) {
-      if (!this.element.getContext) {
+      if (!(this.element || 0).getContext) {
```

Comment:
@aleen42 please revert this change, this.element is not optional.

Reply:
There is an exception when getElementabyId rerurns null.

Reply:
ok

Reply:
@0xfe Please confirm whether need to revert this change.

Reply:
It's cryptic :-) Since this is typescript, you can do: `if (!this.element?.getContext) { .. }` which might be clearer, no?

Reply:
You're right

Reply:
I have amended and pushed it already.